src/Application/Migrations/24_function_ticket_reset.sql

```diff
@@ -0,0 +1,70 @@
+BEGIN;
+
+--param_handle_id is used for the log entry
+CREATE OR REPLACE FUNCTION ticket_reset(param_ticket_id bigint, param_handle_id bigint)
+  RETURNS void AS
+$$
+DECLARE
+  ticket tbl_ticket%rowtype;
+  t tbl_ticket%rowtype;
+  inital_state enum_ticket_state;
+BEGIN
+  --first get the full given ticket
+  SELECT * INTO ticket FROM tbl_ticket where id = param_ticket_id;
+
+  IF ticket IS NULL THEN
+    RAISE WARNING 'ticket with id % does not exist', param_ticket_id;
+    RETURN;
+  END IF;
+
+  --reset the given ticket to inital state
+  inital_state := ticket_state_initial(ticket.project_id, ticket.ticket_type);
+  IF inital_state <> ticket.ticket_state THEN
+    INSERT INTO tbl_log (
+      ticket_id, from_state, to_state, handle_id, event
+      ) VALUES (ticket.id, ticket.ticket_state, inital_state, param_handle_id, 'Action.reset');
+    UPDATE tbl_ticket SET ticket_state=inital_state WHERE id=param_ticket_id;
+  END IF;
+
+  --if given ticket is a meta ticket, reset all childs
+  IF ticket.ticket_type = 'meta'::enum_ticket_type THEN
+    FOR t IN
+    SELECT * FROM tbl_ticket WHERE parent_id=param_ticket_id
+    LOOP
+      inital_state := ticket_state_initial(t.project_id, t.ticket_type);
+      IF inital_state <> t.ticket_state THEN
+        INSERT INTO tbl_log (
+          ticket_id, from_state, to_state, handle_id, event, comment
+          ) VALUES (t.id, t.ticket_state, inital_state, param_handle_id, 'Action.reset', 'Following meta Ticket.');
+        UPDATE tbl_ticket SET ticket_state=inital_state WHERE id=t.id;
+      END IF;
+    END LOOP;
+
+  --if given ticket is an encoding ticket , reset all ticket depends on it by encoding profile
+  ELSEIF ticket.ticket_type = 'encoding'::enum_ticket_type THEN
+    FOR t IN
+    SELECT t2.* FROM tbl_ticket t2
+      JOIN tbl_encoding_profile_version epv2 ON t2.encoding_profile_version_id = epv2.id
+      JOIN (
+        SELECT
+          encoding_profile_all_dependees(epv.encoding_profile_id) AS id,
+          tt.parent_id AS parent_id
+        FROM tbl_ticket tt
+        JOIN tbl_encoding_profile_version epv ON tt.encoding_profile_version_id = epv.id
+        WHERE tt.id = param_ticket_id
+      ) pd ON pd.id = epv2.encoding_profile_id AND t2.parent_id = pd.parent_id
+    LOOP
+      inital_state := ticket_state_initial(t.project_id, t.ticket_type);
+      IF inital_state <> t.ticket_state THEN
+        INSERT INTO tbl_log (
+          ticket_id, from_state, to_state, handle_id, event, comment
+          ) VALUES (t.id, t.ticket_state, inital_state, param_handle_id, 'Action.reset', 'Following encoding ticket.');
+        UPDATE tbl_ticket SET ticket_state=inital_state WHERE id=t.id;
```

Comment:
Try to be consistent with spacing around operators. I personally like spaces before and after them.